src/workerd/api/node/tests/util-nodejs-test.js

```diff
@@ -4894,3 +4943,913 @@ export const testTypes = {
+      value: A,
+      enumerable: false,
+      configurable: true,
+      writable: true,
+    });
+
+    const b = new B('b');
+    assert.strictEqual(b.a(), 'a');
+    assert.strictEqual(b.b(), 'b');
+    assert.strictEqual(b.constructor, B);
+
+    // Two levels of inheritance
+    function C() {
+      B.call(this, 'b');
+      this._c = 'c';
+    }
+    inherits(C, B);
+    C.prototype.c = function () {
+      return this._c;
+    };
+    C.prototype.getValue = function () {
+      return this.a() + this.b() + this.c();
+    };
+
+    assert.strictEqual(C.super_, B);
+
+    const c = new C();
+    assert.strictEqual(c.getValue(), 'abc');
+    assert.strictEqual(c.constructor, C);
+
+    // Inherits can be called after setting prototype properties
+    function D() {
+      C.call(this);
+      this._d = 'd';
+    }
+
+    D.prototype.d = function () {
+      return this._d;
+    };
+    inherits(D, C);
+
+    assert.strictEqual(D.super_, C);
+
+    const d = new D();
+    assert.strictEqual(d.c(), 'c');
+    assert.strictEqual(d.d(), 'd');
+    assert.strictEqual(d.constructor, D);
+
+    // ES6 classes can inherit from a constructor function
+    class E {
+      constructor() {
+        D.call(this);
+        this._e = 'e';
+      }
+      e() {
+        return this._e;
+      }
+    }
+    inherits(E, D);
+
+    assert.strictEqual(E.super_, D);
+
+    const e = new E();
+    assert.strictEqual(e.getValue(), 'abc');
+    assert.strictEqual(e.d(), 'd');
+    assert.strictEqual(e.e(), 'e');
+    assert.strictEqual(e.constructor, E);
+
+    // Should throw with invalid arguments
+    assert.throws(
+      () => {
+        inherits(A, {});
+      },
+      {
+        code: 'ERR_INVALID_ARG_TYPE',
+        name: 'TypeError',
+        message:
+          'The "superCtor.prototype" property must be of type object. ' +
+          'Received undefined',
+      }
+    );
+
+    assert.throws(
+      () => {
+        inherits(A, null);
+      },
+      {
+        code: 'ERR_INVALID_ARG_TYPE',
+        name: 'TypeError',
+        message:
+          'The "superCtor" argument must be of type function. ' +
+          'Received null',
+      }
+    );
+
+    assert.throws(
+      () => {
+        inherits(null, A);
+      },
+      {
+        code: 'ERR_INVALID_ARG_TYPE',
+        name: 'TypeError',
+        message: 'The "ctor" argument must be of type function. Received null',
+      }
+    );
+  },
+};
+
+// parallel/test-util-promisify.js
+export const testPromisify = {
+  async test() {
+    // TODO(gbedford): expectWarning only possible with process.on('warning')
+    // {
+    //   const warningHandler = commonMustNotCall();
+    //   process.on('warning', warningHandler);
+    //   function foo() {}
+    //   foo.constructor = (async () => {}).constructor;
+    //   promisify(foo);
+    //   process.off('warning', warningHandler);
+    // }
+    //
+    // commonExpectWarning(
```

Comment:
We don't implement deprecated tests at workerd, since it will eventually be removed from node.js

Reply:
This tests is more of a test of `process.on('warning')` than a test of deprecated feature, the deprecation is just a message to test the functionality.

If we don't want to ever implement `process.on('warning')` then I think we should remove this rather only then.